balancer_test.go

```diff
@@ -161,10 +161,11 @@ func TestEmptyAddrs(t *testing.T) {
 	// Loop until the above updates apply.
 	for {
 		time.Sleep(10 * time.Millisecond)
-		ctx, _ := context.WithTimeout(context.Background(), 10*time.Millisecond)
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Millisecond)
 		if err := Invoke(ctx, "/foo/bar", &expectedRequest, &reply, cc); err != nil {
 			break
```

Comment:
We need to also call cancel() before break, otherwise the last context will still be leaked.